pkg/webui/console/components/events/utils/uplink-metadata.js

```diff
@@ -0,0 +1,16 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export default gateways =>
+  gateways.reduce((prev, current) => (prev.snr >= current.snr ? prev : current))
```

Comment:
No need to create a new module for this. This can just be added to `console/components/events/utils/index.js`